alphagov/paas-cf
# [#152144657] Solve some of the credentials rotation issues

## What

We need to make sure we don't have downtime and the create-cloudfoundry pipeline runs successfully when we rotate our credentials.

We identified two major issues:
 - the uaa_admin_password is used before, during and after the cf-deploy job which means e.g. the api-availability-test will have a lot of failures during deployment.
 - the uaa postgres user password change will cause downtime in UAA

This PR contains the following changes:
 * Add concourse task for rotating UAA admin credentials: this task will generate a new uaa_admin_password secret and update it in UAA and the cf-secrets.yml.
 * Do not delete uaa_admin_password when clearing the credentials (now the rotate task will handle it)
 * Trigger clear-cloudfoundry-credentials after rotate-cloudfoundry-credentials: this makes the process more clear
 * Don't rotate CF database user passwords as this causes downtime during deployment.

Minor changes:
 * Order cf-secret keys alphabetically for easier diff

## How to review

1. Update your pipelines from this branch
   ```BRANCH=rotate_credentials_201711_152144657 make dev pipelines```
2. Run the rotate-cloudfoundry-credentials job - it should trigger the clear-cloudfoundry-credentials job
3. Run the create-cloudfoundry pipeline.
4. Check the api-availability pipeline for errors. In our tests we only saw one error like this:
   ```"Failed to fetch app stats: Error requesting app stats: cfclient: error (200002): CF-StatsUnavailable": 1,```
5. All other tests should succeed

## Who can review

Not @bandesz
